src/components/tile/tile.stories.tsx

```diff
@@ -161,6 +162,107 @@ export const WithTileFooter: ComponentStory<typeof Tile> = () => (
   </Box>
 );
 
+export const WithTileHeader: ComponentStory<typeof Tile> = () => (
+  <Box>
+    <Tile orientation="vertical" width={400}>
+      <TileContent>
+        <TileHeader variant="transparent" pb={2}>
+          <Typography pr={2} display="inline" variant="b">
+            Example bold text
+          </Typography>
+          <Typography display="inline">Example text</Typography>
+        </TileHeader>
+        <Box pt={2}>
+          <Typography pb={2} variant="h4" fontWeight="bold">
+            Example header
+          </Typography>
+          <Typography>
+            Labore ipsum nostrud quis aliquip esse cillum excepteur commodo
+            tempor. Ex tempor sunt culpa culpa tempor culpa. Laboris dolor nisi
+            ex voluptate occaecat veniam. Magna aliqua velit aliquip dolore
+            pariatur nostrud deserunt amet.
+          </Typography>
+        </Box>
+      </TileContent>
+    </Tile>
+    <Box my={3} />
+    <Tile px={0} pt={0} orientation="vertical" width={400}>
+      <TileContent>
+        <TileHeader p={3}>
+          <Typography pr={2} display="inline" variant="b">
+            Example bold text
+          </Typography>
+          <Typography display="inline">Example text</Typography>
+        </TileHeader>
+        <Box px={3} pt={3}>
+          <Typography pb={2} variant="h4" fontWeight="bold">
+            Example header
+          </Typography>
+          <Typography>
+            Labore ipsum nostrud quis aliquip esse cillum excepteur commodo
+            tempor. Ex tempor sunt culpa culpa tempor culpa. Laboris dolor nisi
+            ex voluptate occaecat veniam. Magna aliqua velit aliquip dolore
+            pariatur nostrud deserunt amet.
+          </Typography>
+        </Box>
+      </TileContent>
+    </Tile>
+    <Tile px={0} pt={0} orientation="vertical" width={400}>
+      <TileContent>
+        <TileHeader p={2} variant="black" />
+        <Box px={3} pt={3}>
+          <Typography pb={2} variant="h4" fontWeight="bold">
+            Example header
+          </Typography>
+          <Typography>
+            Labore ipsum nostrud quis aliquip esse cillum excepteur commodo
+            tempor. Ex tempor sunt culpa culpa tempor culpa. Laboris dolor nisi
+            ex voluptate occaecat veniam. Magna aliqua velit aliquip dolore
+            pariatur nostrud deserunt amet.
+          </Typography>
+        </Box>
+      </TileContent>
+    </Tile>
+  </Box>
+);
+
+export const WithButtonInTileHeader: ComponentStory<typeof Tile> = () => {
```

Comment:
**issue:** The corners appearing to be missing a bit off this example. 

![Screenshot 2024-02-09 at 11 08 48](https://github.com/Sage/carbon/assets/56251247/e030e15d-b7d6-4d11-929c-f2998532475d)